src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs

```diff
@@ -40,7 +40,11 @@ public ServiceManagerBuilder WithCallingAssembly()
         public IServiceManager Build()
         {
             _options.ValidateOptions();
-            return new ServiceManager(_options, ProductInfo.GetProductInfo(_assembly));
+
+            var productInfo = ProductInfo.GetProductInfo(_assembly);
+            ISignalRServiceRestClient restClient = new RestClientBuilder(_options.ConnectionString).WithUserAgent(productInfo).Build();
```

Comment:
sounds that the `RestClientBuilder` can be a local field of `ServiceManagerBuilder`?

Reply:
If RestClientBuilder is a local field, then each change to connection string should trigger a construction of RestClientBuilder, which makes things more complicated.